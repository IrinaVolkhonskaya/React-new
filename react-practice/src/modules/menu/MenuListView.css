.MuiButton-containedSizeSmall {
  padding: 4px 10px;
  font-size: 0.8125rem;
  margin: 0 0 0 17px;
}

.menubutton {
  position: absolute;
  bottom: 0;
}

.ui.five.cards>.card {
  height: 370px;
}

