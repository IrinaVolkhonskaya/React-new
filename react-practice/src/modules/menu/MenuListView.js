import React from "react";
import { Link } from "react-router-dom";
import routes from "../../configs/routes";

import s from "./MenuListView.module.css";
import "./MenuListView.css";
import { Button } from "@material-ui/core";
import { Card } from "semantic-ui-react";

export default function MenuList({ menu, addToCart }) {
  return (
    menu && (
      <Card.Group itemsPerRow={5}>
        {menu.map(({ id, image, name, price }) => (
          <Card key={id} color="blue" raised image={image}>
            <div className="menuList">
              <img className={s.image} src={image} alt="" />
              <p>
                <b className={s.name}>{name}</b>
              </p>
              <div className="menubutton">
                <p className={s.price}>Цена: {price} грн</p>

                <Link to={`${routes.MENU}/${id}`}>
                  <Button
                    className={s.button}
                    variant="contained"
                    color="primary"
                    size="small"
                  >
                    Детальнее
                  </Button>
                </Link>

                <Button
                  className={s.button}
                  variant="contained"
                  color="primary"
                  size="small"
                  onClick={() => addToCart(id)}
                >
                  В корзину
                </Button>
              </div>
            </div>
          </Card>
        ))}
      </Card.Group>
    )
  );
}
