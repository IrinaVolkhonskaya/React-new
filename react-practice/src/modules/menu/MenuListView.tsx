import React from "react";
import { Link } from "react-router-dom";
import routes from "../../configs/routes";

import s from "./MenuListView.module.css";
import "./MenuListView.css";
import { Button } from "@material-ui/core";
import { Card } from "semantic-ui-react";

interface IMenuList{
  menu: IMenuItem[],
  addToCart: (arg0: number) => number,
}

interface IMenuItem {
  id: number,
  image: string,
  name: string, 
  price: number
}

export default function MenuList({ menu, addToCart }: IMenuList) {
  return (
    menu && (
      <Card.Group itemsPerRow={5}>
        {menu.map(({ id, image, name, price }: IMenuItem) => (
          <Card key={id} color="blue" raised image={image}>
            <div className="menuList">
              <img className={s.image} src={image} alt="" />
              <p>
                <b className={s.name}>{name}</b>
              </p>
              <div className="menubutton">
                <p className={s.price}>Цена: {price} грн</p>

                <Link to={`${routes.MENU}/${id}`}>
                  <Button
                    className={s.button}
                    variant="contained"
                    color="primary"
                    size="small"
                  >
                    Детальнее
                  </Button>
                </Link>

                <Button
                  className={s.button}
                  variant="contained"
                  color="primary"
                  size="small"
                  onClick={() => addToCart(id)}
                >
                  В корзину
                </Button>
              </div>
            </div>
          </Card>
        ))}
      </Card.Group>
    )
  );
}
