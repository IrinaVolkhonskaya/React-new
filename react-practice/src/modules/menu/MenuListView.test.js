import React from "react";
import MenuListView from "./MenuListView";
import { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Enzyme from "enzyme";
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

describe("MenuListView component", () => {
    it("should render MenuItemView component", () => {
        const component = shallow(<MenuListView />);
        expect(toJson(component)).toMatchSnapshot();
    });
})