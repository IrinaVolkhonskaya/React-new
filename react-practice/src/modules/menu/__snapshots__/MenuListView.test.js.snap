// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuListView component should render MenuItemView component 1`] = `""`;
